src/logic/map_objects/tribes/militarysite.cc

```diff
@@ -396,7 +396,8 @@ void MilitarySite::update_statistics_string(std::string* s) {
 					cr->push_arg(capacity_);
 					cr->resume();
 					std::string new_string = cr->pop_string();
-					statistics_string_cache_[idx].insert(std::make_pair(stat, new_string));
+					statistics_string_cache_[idx].insert(
+					   std::make_pair((pres << 8) + (stat << 4) + capacity_, new_string));
```

Comment:
Good catch :) 
But wouldn't milsites with large soldier capacities still show the bug if the capacity can occupy more than four bits? Perhaps use std::tuple<int,int,int> as key instead of plain int to ensure all keys are unique.

Reply:
I thought the current milsites don't have more than 12 soldiers. But for future expansions (and Addons) you are right.